nestkernel/event.h

```diff
@@ -827,13 +843,20 @@ class GapJEvent : public SecondaryEvent
   static void
   set_syn_id( const synindex synid )
```

Comment:
This is used in `connector_model_impl.h`

```cpp
/**
 * Register a synape with default Connector and without any common properties.
 */
template < class ConnectionT >
synindex
register_secondary_connection_model( Network& net, const std::string& name, bool has_delay = true )
{
  ConnectorModel& cm =
    *( new GenericSecondaryConnectorModel< ConnectionT >( net, name, has_delay ) );

  synindex synid = net.register_secondary_synapse_prototype( &cm );

  ConnectionT::EventType::set_syn_id( synid );

  return synid;
}
```

At this point it is important, that this is a `static` method. The other method `add_syn_id()` on the other side is called from an object.